pkg/storegateway/sharding_strategy_test.go

```diff
@@ -64,6 +67,21 @@ func TestDefaultShardingStrategy(t *testing.T) {
 				"127.0.0.2": {block2, block4},
 			},
 		},
+		"two ACTIVE instances in the ring with replication factor = 1 and one tenant disabled": {
```

Comment:
Unfortunately, this function does not test FilterUsers, it tests FilterBlocks. So this is not the right place